### PR TITLE
Unimproved time millis spent score difference threshold termination

### PR DESCRIPTION
**Pull request by ge0ffrey**
Docs coming soon, as well as support for 
`
<unimprovedScoreDifferenceThreshold>1hard/MIN_VALUEsoft</unimprovedScoreDifferenceThreshold>`

Tests coming soon too hopefully, the use of System.currentMillis() makes this a pain, but we need it...

### CODE REVIEW

**Comment by ge0ffrey** [comment]
@rsynek Because this fixes the termination lifecycle issue, it might deteriorate performance (probably not, I don't notice it). Any way we can verify/deny that before merging?

**Comment by rsynek** [comment]
@ge0ffrey went over the PR, looks good so far. Regarding the performance testing; please read my comment on Slack.

**Comment by ge0ffrey** [comment]
@rsynek The test for this Unimproved termination is blocked by PLANNER-1270 and I 've added it to that jira that it must be written.

File: optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc [hunk 1]
```diff
@@ -403,18 +425,56 @@ Terminates when the best score has not improved in a specified amount of time.
...
+      <unimprovedSecondsSpentLimit>30</unimprovedSecondsSpentLimit>
+      <unimprovedScoreDifferenceThreshold>0hard/100soft</unimprovedScoreDifferenceThreshold>
+    </termination>
+  </localSearch>
+----
+
+If the score improves by 1 hard point and drops 900 soft points, it's still meets the threshold,
+because `1hard/-900soft` is larger than the threshold `0hard/100soft`.
+
+On the other hand, a threshold of `1hard/0soft` is not met by an new best solution
```
Comment (rsynek) [inline optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc]: "an" -> "any" ?

**Pull request merged**